app/gui/src/controller/searcher.rs

```diff
@@ -1050,28 +1046,36 @@ impl Searcher {
         fragments: impl Iterator<Item = &'a FragmentAddedByPickingSuggestion>,
         permanent: bool,
     ) -> FallibleResult {
-        let imports = fragments.flat_map(|frag| self.code_to_insert(frag).imports);
+        let imports = fragments.flat_map(|frag| {
+            frag.required_imports(&self.database, self.module_qualified_name().as_ref())
+        });
         let mut module = self.module();
-        let here = self.module_qualified_name();
         // TODO[ao] this is a temporary workaround. See [`Searcher::add_enso_project_entries`]
         //     documentation.
```

Comment:
The referenced docs link to an obsolete issue. How temporary is this?

Reply:
Hard to say. I think the vision how to handle that might changed. Discussion continues here: https://discordapp.com/channels/401396655599124480/1034469032788762734/1034469032788762734